src/main/java/io/github/pascalklassen/pokefuture/PokemonService.java

```diff
@@ -17,22 +66,90 @@
+        caches.put(BerryFlavor.class, createCache());
+        caches.put(Characteristic.class, createCache());
+        caches.put(ContestType.class, createCache());
+        caches.put(ContestEffect.class, createCache());
+        caches.put(EggGroup.class, createCache());
+        caches.put(EncounterMethod.class, createCache());
+        caches.put(EncounterCondition.class, createCache());
+        caches.put(EncounterConditionValue.class, createCache());
+        caches.put(EvolutionChain.class, createCache());
+        caches.put(EvolutionTrigger.class, createCache());
+        caches.put(Generation.class, createCache());
+        caches.put(Gender.class, createCache());
+        caches.put(GrowthRate.class, createCache());
+
+        caches.put(Item.class, createCache());
+        caches.put(ItemCategory.class, createCache());
+        caches.put(ItemAttribute.class, createCache());
+        caches.put(ItemFlingEffect.class, createCache());
+        caches.put(ItemPocket.class, createCache());
+        caches.put(Language.class, createCache());
+        caches.put(Location.class, createCache());
+        caches.put(LocationArea.class, createCache());
+        caches.put(Move.class, createCache());
+        caches.put(MoveAilment.class, createCache());
+        caches.put(MoveBattleStyle.class, createCache());
+        caches.put(MoveCategory.class, createCache());
+        caches.put(MoveDamageClass.class, createCache());
+        caches.put(MoveLearnMethod.class, createCache());
+        caches.put(MoveTarget.class, createCache());
+        caches.put(Nature.class, createCache());
+
+        caches.put(PalParkArea.class, createCache());
+        caches.put(Pokedex.class, createCache());
+        caches.put(Pokemon.class, createCache());
+        caches.put(PokemonColor.class, createCache());
+        caches.put(PokemonForm.class, createCache());
+        caches.put(PokemonHabitat.class, createCache());
+        caches.put(PokemonShape.class, createCache());
+        caches.put(PokemonSpecies.class, createCache());
+        caches.put(PokeathlonStat.class, createCache());
+        caches.put(Region.class, createCache());
+        caches.put(Stat.class, createCache());
+        caches.put(SuperContestEffect.class, createCache());
+        caches.put(Type.class, createCache());
+        caches.put(Version.class, createCache());
+        caches.put(VersionGroup.class, createCache());
+
+    }
+
     private PokemonService() {
     }
 
+    private static Cache<String, Object> createCache(){
+        return CacheBuilder.newBuilder().expireAfterAccess(10, TimeUnit.MINUTES).build();
+    }
+
     public static <T> Future<T> fetchAs(@NotNull Class<T> clazz, @NotNull String uri) {
         return fetchAs(clazz, uri, false);
     }
 
     public static <T> Future<T> fetchAs(@NotNull Class<T> clazz, @NotNull String uri, boolean absolute) {
         Promise<T> promise = Promise.promise();
 
+        if(isCache && getValue(clazz, uri) != null){
```

Comment:
### Erstens
Bei allen ControlFlow Elementen, also `if`, `switch`, `for` etc bitte ein Leerzeichen zwischen dem Wort und der Klammer UND der gewölbten Klammer.

Beispiel:
```java
// Richtig
if (true) {
}

// Falsch!
if(true){
}
```

### Zweitens
Speicher am besten den Wert von `getValue` in einer variable davor ab und frag einfach ab ob diese variable null ist anstatt zweimal `getValue` aufzurufen. Das mag zwar keinen riesen Unterschied machen aber wenn hier tausende Anfragen geschehen dürfte das ein paar Millisekunden wettmachen^^